server/src/collection.rs

```diff
@@ -75,11 +75,60 @@ impl Collection {
     #[zbus(out_args("item", "prompt"))]
     pub async fn create_item(
         &self,
-        _properties: Properties,
-        _secret: SecretInner,
-        _replace: bool,
-    ) -> Result<(OwnedObjectPath, ObjectPath), ServiceError> {
-        todo!()
+        properties: Properties,
+        secret: SecretInner,
+        replace: bool,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+        #[zbus(signal_emitter)] signal_emitter: zbus::object_server::SignalEmitter<'_>,
+    ) -> Result<(OwnedObjectPath, OwnedObjectPath), ServiceError> {
+        let SecretInner(session, iv, secret, _content_type) = secret;
+        let label = properties.label();
+        let attributes = properties.attributes().unwrap();
+        let manager = self.manager.lock().await;
+
+        let Some(session) = manager.session(&session) else {
+            tracing::error!("The session `{}` does not exist.", session);
+            return Err(ServiceError::NoSession(format!(
+                "The session `{}` does not exist.",
+                session
+            )));
+        };
+
+        let secret = match session.aes_key() {
+            Some(key) => oo7::crypto::decrypt(secret, &key, &iv),
+            None => zeroize::Zeroizing::new(secret),
+        };
+
+        let item = match self
+            .keyring
+            .create_item(label, &attributes, secret, replace)
+            .await
+        {
+            Ok(item) => item,
+            Err(_) => todo!(), // TODO: may be new ServiceError
+        };
+
+        let n_items = *self.item_index.read().await;
+        let item = item::Item::new(
+            item,
+            self.is_locked().await,
+            Arc::clone(&self.manager),
+            &self.path,
+            n_items,
+        );
+        *self.item_index.write().await = n_items + 1;
+
+        self.items.lock().await.push(item.clone());
+
+        Self::item_created(&signal_emitter, item.path()).await?;
+        self.items_changed(&signal_emitter).await?;
```

Comment:
should be emitted after the item is served.

Reply:
Done, thanks!